attestation-service/src/token/ear_default_policy.rego

```diff
@@ -0,0 +1,132 @@
+
+default se_configuration := 36
+
+executables := min({sample_executables, snp_executables, tdx_executables, az_snp_executables, az_tdx_executables, se_executables})
+
+hardware := min({sample_hardware, snp_hardware, tdx_hardware, az_snp_hardware, az_tdx_hardware, se_hardware})
+
+configuration := min({sample_configuration, snp_configuration, tdx_configuration, az_snp_configuration, az_tdx_configuration, se_configuration})
+
+##### Sample
+
+# For the `executables` trust claim, the value 3 stands for
+# "Only a recognized genuine set of approved executables have
+#  been loaded during the boot process."
+sample_executables := 3 if {
+	# The sample attester does not report any launch digest.
+	# This is an example of how a real platform might validate executables.
+	input.sample.launch_digest in data.reference.launch_digest
+}
+
+# For the `hardware` trust claim, the value 2 stands for
+# "An Attester has passed its hardware and/or firmware
+#  verifications needed to demonstrate that these are genuine/
+#  supported.
+sample_hardware := 2 if {
+	# The sample attester does not report any productId.
+	# This is an exmple of how a real platform might identify the hardware
+	# that is running.
+	input.sample.productId in data.reference.productId
+	input.sample.svn in data.reference.svn
+}
+
+##### SNP
+snp_executables := 3 if {
+	# In the future, we might calculate this measurement here various components
+	input.sample.launch_measurement in data.reference.snp_launch_measurement
+}
+
+snp_hardware := 2 if {
+	# Check the reported TCB to validate the ASP FW
+	input.snp.reported_tcb_bootloader in data.reference.snp_bootloader
+	input.snp.reported_tcb_microcode in data.reference.snp_microcode
+	input.snp.reported_tcb_snp in data.reference.snp_snp_svn
+	input.snp.reported_tcb_tee in data.reference.snp_tee_svn
+}
+
+# For the 'configuration' trust claim 2 stands for
+# "The configuration is a known and approved config."
+#
+# For this, we compare all the configuration fields.
+snp_configuration := 2 if {
+	input.snp.policy_debug_allowed == 0
+	input.snp.policy_migrate_ma == 0
+	input.snp.platform_smt_enabled in data.reference.snp_smt_enabled
+	input.snp.platform_tsme_enabled in data.reference.snp_tsme_enabled
+	input.snp.policy_abi_major in data.reference.snp_guest_abi_major
+	input.snp.policy_abi_minor in data.reference.snp_guest_abi_minor
+	input.snp.policy_single_socket in data.reference.snp_single_socket
+	input.snp.policy_smt_allowed in data.reference.snp_smt_allowed
+}
+
+# For the `configuration` trust claim 3 stands for
+# "The configuration includes or exposes no known
+#  vulnerabilities."
+#
+# In this check, we do not specifically check every
+# configuration value, but we make sure that some key
+# configurations (like debug_allowed) are set correctly.
+else := 3 if {
+	input.snp.policy_debug_allowed == 0
+	input.snp.policy_migrate_ma == 0
+}
+
+##### TDX TODO
+##### AZ SNP TODO
+##### AZ TDX TODO
+##### SE TODO
```

Comment:
We will still get an EAR token but it will contain a contraindicated trust vector. Since we don't actually check the policy result in our existing attestation token, this shouldn't matter too much in the short term.

I am planning to create a follow-up PR that adds tests that actually check the TCB and that use more interesting KBS policies. Hopefully we can collaboratively iterate on the AS policy after that with some of the HW experts.